cloudinit/config/schemas/schema-network-config-v1.json

```diff
@@ -0,0 +1,562 @@
+            },
+            "mac_address": {
+              "type": "string",
+              "description": "The lowercase MAC address of the physical device."
+            },
+            "mtu": {
+              "type": "integer",
+              "description": "The MTU size in bytes. The ``mtu`` key represents a device's Maximum Transmission Unit, which is the largest size packet or frame, specified in octets (eight-bit bytes), that can be sent in a packet- or frame-based network. Specifying ``mtu`` is optional. Values too small or too large for a device may be ignored by that device."
+            },
+            "subnets": {
+              "type": "array",
+              "items": {
+                  "$ref": "#/$defs/config_type_subnet"
+              }
+            }
+          }
+        },
+        "anyOf_type_bond": {
+          "type": "object",
+          "additionalProperties": false,
+          "required": ["type", "name"],
+          "properties": {
+            "type": {
+              "type": "string",
+              "enum": ["bond"]
+            },
+            "name": {
+              "type": "string",
+              "description": "Desired device name should be less than 15 characters. Any characters exceeding 15 will be truncated. This is a limitation of the Linux kernel network-device structure."
+            },
+            "mac_address": {
+              "type": "string",
+              "description": "When specifying MAC Address on a bond this value will be assigned to the bond device and may be different than the MAC address of any of the underlying bond interfaces. Specifying a MAC Address is optional. If ``mac_address`` is not present, then the bond will use one of the MAC Address values from one of the bond interfaces"
+            },
+            "bond_interfaces": {
+              "type": "array",
+              "description": "The list of network device ``name``s associated with this bond. This list may be empty.",
+              "items": {
+                "type": "string"
+              }
+            },
+            "mtu": {
+              "type": "integer",
+              "description": "The MTU size in bytes. This ``mtu`` key represents a device's Maximum Transmission Unit, which is the largest size packet or frame, specified in octets (eight-bit bytes), that can be sent in a packet- or frame-based network. Specifying ``mtu`` is optional. Values too small or too large for a device may be ignored by that device."
+            },
+            "params": {
+              "desciption": "The ``params`` key in a bond holds a dictionary of bonding parameters. This dictionary may be empty. For more details on what the various bonding parameters mean please read the Linux Kernel :file:`Bonding.txt`.",
+              "additionalProperties": false,
+              "properties":{
```

Comment:
Our unittests have both [hyphenated](https://github.com/canonical/cloud-init/blob/main/tests/unittests/test_net.py#L170) and [underscore delimited](https://github.com/canonical/cloud-init/blob/main/tests/unittests/test_net.py#L2827) key names sprinkled throughout test_net.py.  We also have a number of previous ubuntu-sru test validation scripts verifying that both hyphenated and underscore-delimited keys are honored.
https://github.com/cloud-init/ubuntu-sru/blob/main/bugs/lp-1784699.txt#L39
bugs/lp-1690480/network_data.json.

This list is likely stale from a few years ago and I pulled the initial list form our advertised docs  [for network v1 ](https://cloudinit.readthedocs.io/en/latest/reference/network-config-format-v1.html#bond). I thought we'd start with what we published in docs and grow it as needed to reflect additional/newer key/config changes.

Reply:
I'm a little wary of the day warnings become errors as any bonding param would have worked here, and it may turn into us needlessly restricting it and throwing errors for valid options. I think this is fine as-is, but I think it's worth thinking about what we'll need to change between now and then.